src/main/java/cd/go/plugin/config/yaml/transforms/DefaultOverrides.java

```diff
@@ -51,4 +89,78 @@ public static void overrideGitMaterialAutoUpdateInverse(Map<String, Object> yaml
 			}
 		}
 	}
+
+	public static void addApprovalManualForPullRequests(JsonObject pipeline) {
+		if (!isApprovalManualForPRsFullyConfigured()) {
+			return;
+		}
+
+		JsonArray materials = pipeline.getAsJsonArray(JSON_PIPELINE_MATERIALS_FIELD);
+		if (materials == null) {
+			return;
+		}
+
+		boolean hasPRMaterials = JSONUtils.stream(materials).anyMatch(DefaultOverrides::isPRMaterial);
+		if (!hasPRMaterials) {
+			return;
+		}
+
+		JsonArray stages = pipeline.getAsJsonArray(JSON_PIPELINE_STAGES_FIELD);
+		if (stages == null || stages.size() == 0) {
+			return;
+		}
+		JsonObject firstStage = stages.get(0).getAsJsonObject();
+		if (firstStage.has("approval")) {
+			return;
+		}
+		JsonObject approvalJson = new JsonObject();
+		approvalJson.addProperty(JSON_STAGE_APPROVAL_TYPE_FIELD, "manual");
+		firstStage.add(JSON_STAGE_APPROVAL_FIELD, approvalJson);
+	}
+
+	public static void addApprovalManualForPullRequestsInverse(Map<String, Object> pipeline) {
```

Comment:
What does inverse mean here? 

Maybe my question is: why is the pipeline sometimes a Map, sometimes a JsonObject?

Reply:
There's a feature to export a pipeline to the config repo format (https://api.gocd.org/current/#export-pipeline-config-to-config-repo-format). E.g. you can configure your pipeline in the GUI, then export it to yaml. As far as I understand the inverse transform is for that: transforming the json format of GoCD to the yaml format of the plugin.

We don't use the feature, but for correctness I implemented the inverse as well.